src/modules/ud/components/content-meta-preview/content.meta.preview.js

```diff
@@ -1,7 +1,44 @@
-import React from 'react';
+import React, { useContext, useMemo } from 'react';
+
+import { MarkedLocationContext, LoadedLocationsMapContext } from '../../universal.discovery.module';
 
 const ContentMetaPreview = () => {
-    return <div className="c-content-meta-preview">content meta</div>;
+    const [markedLocation, setMarkedLocation] = useContext(MarkedLocationContext);
+    const [loadedLocationsMap, dispatchLoadedLocationsAction] = useContext(LoadedLocationsMapContext);
+    const { formatShortDateTime } = window.eZ.helpers.timezone;
+    const location = useMemo(() => {
+        return loadedLocationsMap.reduce((item, loadedLocation) => {
+            return loadedLocation.items.find((childrenLocation) => childrenLocation.id === markedLocation) || item;
```

Comment:
Swapping `find` with `item` should improve performance:
`item || loadedLocation.items.find((childrenLocation) => childrenLocation.id === markedLocation)`
we would search for item only if previously nothing was found.